File: backoffice/src/shared/domain/IntValueObject.py
"""
 *
 * Classes 
 *
"""

class IntValueObject:

    """
     *
     * Parameters 
     *
    """

    __value : int

    """
     *
     * Methods 
     *
    """

    def __init__( self, value : int ) -> None:
        self.__value = value
    
    def value( self ) -> int:
        return self.__value
    
    def equals( self, value : int ) -> bool:
        if self.__value == value:
            return True
        return False
    
    def isLessThan( self, value : int ) -> bool:
        if self.__value < value:
            return True
        return False
    
    def isBiggerThan( self, value : int ) -> bool:
        if self.__value > value:
            return True
        return False
    
    def toString( self ) -> str:
        return str( self.__value )

File: attention/src/shared/domain/valueObject/IntValueObject.cs
/* 
 * 
 * Classes 
 *
*/

namespace attention.src.shared.domain {

    public class IntValueObject {

        /* 
         * 
         * Attributes
         *
        */

        private int _value;

        /* 
         * 
         * Methods
         *
        */

        public IntValueObject( int value ) {
            _value = value;
        }

        public int value() {
            return _value;
        }

        public bool equals( int value ) {
            return _value == value;
        }

        public bool isLessThan( int value ) {
            return _value < value;
        }

        public bool isBiggerThan( int value ) {
            return _value > value;
        }

        public string toString() {
            return _value.ToString();
        }

    }

}